server.go

```diff
@@ -0,0 +1,165 @@
+package discoverkit
+
+import (
+	"sync"
+
+	"github.com/dogmatiq/configkit"
+	"github.com/dogmatiq/interopspec/discoverspec"
+)
+
+// Server is an implementation of discoverspec.DiscoverAPIServer.
+type Server struct {
+	m sync.Mutex
+
+	// available is the set of applications that are currently available indexed
+	// by their identity key.
+	//
+	// The map value itself is treated as though it is immutable. When a change
+	// to the available applications is made the map is cloned and the changes
+	// applied to the clone. Finally, s.available is updated to refer to the
+	// clone.
```

Comment:
```suggestion
	// applied to the clone. Finally, the available field is updated to refer
	// to the clone.
```

Reply:
@koden-km how about this ^^?

Reply:
That seems alright. The other one with the `s.` wasn't _bad_ it just seemed like it might have been a copy paste so I thought I should mention it.